File: test-services/java/template/src/main/java/theservice/services/EchoServiceImpl.java
package theservice.services;

import java.math.BigDecimal;
import java.time.*;
import java.util.*;

{{#server.micronaut}}
import io.micronaut.context.annotation.Bean;
{{/server.micronaut}}
{{#server.spring}}
import org.springframework.stereotype.Service;
{{/server.spring}}

import theservice.models.*;
import theservice.services.echo.*;

{{#server.micronaut}}
@Bean
{{/server.micronaut}}
{{#server.spring}}
@Service("EchoService")
{{/server.spring}}
public class EchoServiceImpl implements EchoService {
	@Override
	public String echoBodyString(String body) {
		return body;
	}

	@Override
	public Message echoBodyModel(Message body) {
		return body;
	}

	@Override
	public List<String> echoBodyArray(List<String> body) {
		return body;
	}

	@Override
	public Map<String, String> echoBodyMap(Map<String, String> body) {
		return body;
	}

	@Override
	public Parameters echoQuery(int intQuery, long longQuery, float floatQuery, double doubleQuery, BigDecimal decimalQuery, boolean boolQuery, String stringQuery, String stringOptQuery, String stringDefaultedQuery, List<String> stringArrayQuery, UUID uuidQuery, LocalDate dateQuery, List<LocalDate> dateArrayQuery, LocalDateTime datetimeQuery, Choice enumQuery) {
		return new Parameters(intQuery, longQuery, floatQuery, doubleQuery, decimalQuery, boolQuery, stringQuery, stringOptQuery, stringDefaultedQuery, stringArrayQuery, uuidQuery, dateQuery, dateArrayQuery, datetimeQuery, enumQuery);
	}

	@Override
	public Parameters echoHeader(int intHeader, long longHeader, float floatHeader, double doubleHeader, BigDecimal decimalHeader, boolean boolHeader, String stringHeader, String stringOptHeader, String stringDefaultedHeader, List<String> stringArrayHeader, UUID uuidHeader, LocalDate dateHeader, List<LocalDate> dateArrayHeader, LocalDateTime datetimeHeader, Choice enumHeader) {
		return new Parameters(intHeader, longHeader, floatHeader, doubleHeader, decimalHeader, boolHeader, stringHeader, stringOptHeader, stringDefaultedHeader, stringArrayHeader, uuidHeader, dateHeader, dateArrayHeader, datetimeHeader, enumHeader);
	}

	@Override
	public UrlParameters echoUrlParams(int intUrl, long longUrl, float floatUrl, double doubleUrl, BigDecimal decimalUrl, boolean boolUrl, String stringUrl, UUID uuidUrl, LocalDate dateUrl, LocalDateTime datetimeUrl, Choice enumUrl) {
		return new UrlParameters(intUrl, longUrl, floatUrl, doubleUrl, decimalUrl, boolUrl, stringUrl, uuidUrl, dateUrl, datetimeUrl, enumUrl);
	}

	@Override
	public EchoEverythingResponse echoEverything(Message body, float floatQuery, boolean boolQuery, UUID uuidHeader, LocalDateTime datetimeHeader, LocalDate dateUrl, BigDecimal decimalUrl) {
		return new EchoEverythingResponse.Ok(new Everything(body, floatQuery, boolQuery, uuidHeader, datetimeHeader, dateUrl, decimalUrl));
	}

	@Override
	public SameOperationNameResponse sameOperationName() {
		return new SameOperationNameResponse.Ok();
	}
}

File: test-services/kotlin/template/src/main/kotlin/theservice/services/EchoServiceImpl.kt
package theservice.services

{{#server.micronaut}}
import io.micronaut.context.annotation.Bean
{{/server.micronaut}}
{{#server.spring}}
import org.springframework.stereotype.Service
{{/server.spring}}
import theservice.models.*
import java.math.BigDecimal
import java.util.UUID
import java.time.*
import theservice.services.echo.*

{{#server.micronaut}}
@Bean
{{/server.micronaut}}
{{#server.spring}}
@Service("EchoService")
{{/server.spring}}
class EchoServiceImpl : EchoService {
    override fun echoBodyString(body: String): String {
        return body
    }

    override fun echoBodyModel(body: Message): Message {
        return body
    }

    override fun echoBodyArray(body: List<String>): List<String> {
        return body
    }

    override fun echoBodyMap(body: Map<String, String>): Map<String, String> {
        return body
    }

//    override fun echoFormData(
//        intParam: Int,
//        longParam: Long,
//        floatParam: Float,
//        doubleParam: Double,
//        decimalParam: BigDecimal,
//        boolParam: Boolean,
//        stringParam: String,
//        stringOptParam: String?,
//        stringDefaultedParam: String,
//        stringArrayParam: List<String>,
//        uuidParam: UUID,
//        dateParam: LocalDate,
//        dateArrayParam: List<LocalDate>,
//        datetimeParam: LocalDateTime,
//        enumParam: Choice
//    ): FormParameters {
//        return FormParameters(
//            intParam,
//            longParam,
//            floatParam,
//            doubleParam,
//            decimalParam,
//            boolParam,
//            stringParam,
//            stringOptParam,
//            stringDefaultedParam,
//            stringArrayParam,
//            uuidParam,
//            dateParam,
//            dateArrayParam,
//            datetimeParam,
//            enumParam
//        )
//    }
//
//    override fun echoFormUrlencoded(
//        intParam: Int,
//        longParam: Long,
//        floatParam: Float,
//        doubleParam: Double,
//        decimalParam: BigDecimal,
//        boolParam: Boolean,
//        stringParam: String,
//        stringOptParam: String?,
//        stringDefaultedParam: String,
//        stringArrayParam: List<String>,
//        uuidParam: UUID,
//        dateParam: LocalDate,
//        dateArrayParam: List<LocalDate>,
//        datetimeParam: LocalDateTime,
//        enumParam: Choice
//    ): FormParameters {
//        return FormParameters(
//            intParam,
//            longParam,
//            floatParam,
//            doubleParam,
//            decimalParam,
//            boolParam,
//            stringParam,
//            stringOptParam,
//            stringDefaultedParam,
//            stringArrayParam,
//            uuidParam,
//            dateParam,
//            dateArrayParam,
//            datetimeParam,
//            enumParam
//        )
//    }

    override fun echoQuery(
        intQuery: Int,
        longQuery: Long,
        floatQuery: Float,
        doubleQuery: Double,
        decimalQuery: BigDecimal,
        boolQuery: Boolean,
        stringQuery: String,
        stringOptQuery: String?,
        stringDefaultedQuery: String,
        stringArrayQuery: List<String>,
        uuidQuery: UUID,
        dateQuery: LocalDate,
        dateArrayQuery: List<LocalDate>,
        datetimeQuery: LocalDateTime,
        enumQuery: Choice
    ): Parameters {
        return Parameters(
            intQuery,
            longQuery,
            floatQuery,
            doubleQuery,
            decimalQuery,
            boolQuery,
            stringQuery,
            stringOptQuery,
            stringDefaultedQuery,
            stringArrayQuery,
            uuidQuery,
            dateQuery,
            dateArrayQuery,
            datetimeQuery,
            enumQuery
        )
    }

    override fun echoHeader(
        intHeader: Int,
        longHeader: Long,
        floatHeader: Float,
        doubleHeader: Double,
        decimalHeader: BigDecimal,
        boolHeader: Boolean,
        stringHeader: String,
        stringOptHeader: String?,
        stringDefaultedHeader: String,
        stringArrayHeader: List<String>,
        uuidHeader: UUID,
        dateHeader: LocalDate,
        dateArrayHeader: List<LocalDate>,
        datetimeHeader: LocalDateTime,
        enumHeader: Choice
    ): Parameters {
        return Parameters(
            intHeader,
            longHeader,
            floatHeader,
            doubleHeader,
            decimalHeader,
            boolHeader,
            stringHeader,
            stringOptHeader,
            stringDefaultedHeader,
            stringArrayHeader,
            uuidHeader,
            dateHeader,
            dateArrayHeader,
            datetimeHeader,
            enumHeader
        )
    }

    override fun echoUrlParams(
        intUrl: Int,
        longUrl: Long,
        floatUrl: Float,
        doubleUrl: Double,
        decimalUrl: BigDecimal,
        boolUrl: Boolean,
        stringUrl: String,
        uuidUrl: UUID,
        dateUrl: LocalDate,
        datetimeUrl: LocalDateTime,
        enumUrl: Choice
    ): UrlParameters {
        return UrlParameters(
            intUrl,
            longUrl,
            floatUrl,
            doubleUrl,
            decimalUrl,
            boolUrl,
            stringUrl,
            uuidUrl,
            dateUrl,
            datetimeUrl,
            enumUrl
        )
    }

    override fun echoEverything(
        body: Message,
        floatQuery: Float,
        boolQuery: Boolean,
        uuidHeader: UUID,
        datetimeHeader: LocalDateTime,
        dateUrl: LocalDate,
        decimalUrl: BigDecimal
    ): EchoEverythingResponse {
        return EchoEverythingResponse.Ok(
            Everything(
                body,
                floatQuery,
                boolQuery,
                uuidHeader,
                datetimeHeader,
                dateUrl,
                decimalUrl
            )
        )
    }

    override fun sameOperationName(): SameOperationNameResponse {
        return SameOperationNameResponse.Ok()
    }
}
